src/blockchain/electrum.rs

```diff
@@ -48,6 +48,63 @@ pub struct ElectrumBlockchain {
     stop_gap: usize,
 }
 
+impl IndexedChain for ElectrumBlockchain {
```

Comment:
Since each of these queries requires a different underlying index I think it makes sense to have a different trait for each index e.g. `GetBlockHeader`, `GetPositionInBlock` etc.